### PR TITLE
KDE formulas: remove "kde-" prefix 

### PR DESCRIPTION
**Pull request by wrobelda**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

As discussed in https://github.com/Homebrew/homebrew-core/pull/73032#discussion_r594283101 , I am submitting a PR to rename KDE formulas.

### CODE REVIEW

**Comment by wrobelda** [comment]
I am seeing "New formulae in homebrew/core should not have a `bottle do` block". This isn't a new formula, per se. Should I remove it?  https://docs.brew.sh/Rename-A-Formula doesn't mention anything about it.

**Comment by carlocab** [comment]
> Should I remove it?

You should, yes. Please split your commits so that there is only one formula file modified per commit.

**Comment by wrobelda** [comment]
> You should, yes. Please split your commits so that there is only one formula file modified per commit.

The other files modified update the formulas that reference `kde-extra-cmake-modules`. Do I even have to update them at all? Or is it optional at this point and expected to be done by whomever gets to touch them next time?

File: formula_renames.json [hunk 1]
```diff
@@ -131,5 +131,10 @@
   "v8-315": "v8@3.15",
   "wires": "geckodriver",
   "wpcli-completion": "wp-cli-completion",
-  "xu4": "scummvm"
+  "xu4": "scummvm",
+  "kde-extra-cmake-modules": "extra-cmake-modules",
+  "kde-karchive": "karchive",
+  "kde-ki18n": "ki18n",
+  "kde-kdoctools": "kdoctools",
+  "kde-threadweaver": "threadweaver"
```
Comment (carlocab) [inline formula_renames.json:139]: The rest of this list is sorted alphabetically, can we stick to that?

File: Formula/extra-cmake-modules.rb [hunk 2]
```diff
@@ -1,4 +1,4 @@
-class KdeExtraCmakeModules < Formula
+class ExtraCmakeModules < Formula
```
Comment (carlocab) [inline Formula/extra-cmake-modules.rb:1]: This one seems a little ambiguous. What do other distros call this package? Can we call it `KextraCmakeModules` maybe?

Comment (wrobelda) [reply Formula/extra-cmake-modules.rb:1]: I don't think they're ambiguous, they're not KDE-specific and contain many useful CMake modules. From the project's own description: _"While the driving force of this module is to reduce duplication in CMake scripts across KDE software, it is intended to be useful for any software that uses the CMake build system."_

Also, the vast majority calls it exactly as the upstream does: https://repology.org/project/extra-cmake-modules/versions

File: formula_renames.json (hunk 1 above)
Comment (wrobelda) [reply formula_renames.json:139]: Don't know how I missed that. Fixed.

File: Formula/extra-cmake-modules.rb (hunk 2 above)
Comment (carlocab) [reply Formula/extra-cmake-modules.rb:1]: It's a little ambiguous in that the name makes it sound like the formula is for software from KitWare, but it isn't really. However, if that's what other distros call it, then that's fine.

Comment (wrobelda) [reply Formula/extra-cmake-modules.rb:1]: For what it's worth, the alternative, common description of the project is _"Extra CMake Modules, or ECM, aims to augment CMake with additional modules. It serves as both a staging ground for new modules before they are moved upstream to CMake and a place for modules that, for whatever reason, are not a good fit for CMake itself."_

Many of the ECM modules/improvements were eventually migrated to the CMake itself, and it is a de-facto extension of the CMake itself.

**Review by carlocab — APPROVED** [review]
Thanks, @wrobelda.

**Pull request closed without merging**